src/entities/mixedRoute.test.ts

```diff
@@ -0,0 +1,290 @@
+      expect(route.protocol).toEqual(Protocol.V3)
+    })
+
+    it('inherits parameters from extended route class', () => {
+      const routeOriginal = new V3RouteSDK([pool_0_1], token0, token1)
+      const route = new RouteV3(routeOriginal)
+      expect(route.pools).toEqual(routeOriginal.pools)
+      expect(route.path).toEqual(routeOriginal.tokenPath)
+      expect(route.input).toEqual(routeOriginal.input)
+      expect(route.output).toEqual(routeOriginal.output)
+      expect(route.midPrice).toEqual(routeOriginal.midPrice)
+      expect(route.chainId).toEqual(routeOriginal.chainId)
+    })
+
+    it('can have a token as both input and output', () => {
+      const routeOriginal = new V3RouteSDK([pool_0_weth, pool_0_1, pool_1_weth], weth, weth)
+      const route = new RouteV3(routeOriginal)
+      expect(route.pools).toEqual([pool_0_weth, pool_0_1, pool_1_weth])
+      expect(route.input).toEqual(weth)
+      expect(route.output).toEqual(weth)
+    })
+
+    it('supports ether input', () => {
+      const routeOriginal = new V3RouteSDK([pool_0_weth], ETHER, token0)
+      const route = new RouteV3(routeOriginal)
+      expect(route.pools).toEqual([pool_0_weth])
+      expect(route.input).toEqual(ETHER)
+      expect(route.output).toEqual(token0)
+    })
+
+    it('supports ether output', () => {
+      const routeOriginal = new V3RouteSDK([pool_0_weth], token0, ETHER)
+      const route = new RouteV3(routeOriginal)
+      expect(route.pools).toEqual([pool_0_weth])
+      expect(route.input).toEqual(token0)
+      expect(route.output).toEqual(ETHER)
+    })
+  })
+
+  describe('#midPrice', () => {
+    const pool_0_1 = new Pool(
+      token0,
+      token1,
+      FeeAmount.MEDIUM,
+      encodeSqrtRatioX96(1, 5),
+      0,
+      TickMath.getTickAtSqrtRatio(encodeSqrtRatioX96(1, 5)),
+      []
+    )
+    const pool_1_2 = new Pool(
+      token1,
+      token2,
+      FeeAmount.MEDIUM,
+      encodeSqrtRatioX96(15, 30),
+      0,
+      TickMath.getTickAtSqrtRatio(encodeSqrtRatioX96(15, 30)),
+      []
+    )
+    const pool_0_weth = new Pool(
+      token0,
+      weth,
+      FeeAmount.MEDIUM,
+      encodeSqrtRatioX96(3, 1),
+      0,
+      TickMath.getTickAtSqrtRatio(encodeSqrtRatioX96(3, 1)),
+      []
+    )
+    const pool_1_weth = new Pool(
+      token1,
+      weth,
+      FeeAmount.MEDIUM,
+      encodeSqrtRatioX96(1, 7),
+      0,
+      TickMath.getTickAtSqrtRatio(encodeSqrtRatioX96(1, 7)),
+      []
+    )
+
+    const pair_0_1 = new Pair(CurrencyAmount.fromRawAmount(token0, '100'), CurrencyAmount.fromRawAmount(token1, '200'))
+    const pair_1_2 = new Pair(CurrencyAmount.fromRawAmount(token1, '200'), CurrencyAmount.fromRawAmount(token2, '150'))
+    const pair_0_weth = new Pair(CurrencyAmount.fromRawAmount(token0, '100'), CurrencyAmount.fromRawAmount(weth, '100'))
+    const pair_1_weth = new Pair(CurrencyAmount.fromRawAmount(token1, '175'), CurrencyAmount.fromRawAmount(weth, '100'))
```

Comment:
In the RouteV3 tests, they recreate the pool and pairs here before the midPrice tests. I figured I'd keep that pattern for now, but a benefit would be that we can play with values here for more intricate midPrice tests. A drawback is that we could reduce the amount of code by just using the previously defined ones, and since the var names are the same it can be confusing.